pingen/models/pingen_document.py

```diff
@@ -0,0 +1,445 @@
+            ("error", "Connection Error"),
+            ("pingen_error", "Pingen Error"),
+            ("canceled", "Canceled"),
+        ],
+        readonly=True,
+        required=True,
+        default="pending",
+    )
+    auto_send = fields.Boolean(
+        help="Defines if a document is merely uploaded or also sent",
+        default=True,
+    )
+    delivery_product = fields.Selection(
+        [
+            ("fast", "fast"),
+            ("cheap", "cheap"),
+            ("bulk", "bulk"),
+            ("track", "track"),
+            ("sign", "sign"),
+            ("atpost_economy", "atpost_economy"),
+            ("atpost_priority", "atpost_priority"),
+            ("postag_a", "postag_a"),
+            ("postag_b", "postag_b"),
+            ("postag_b2", "postag_b2"),
+            ("postag_registered", "postag_registered"),
+            ("postag_aplus", "postag_aplus"),
+            ("dpag_standard", "dpag_standard"),
+            ("dpag_economy", "dpag_economy"),
+            ("indpost_mail", "indpost_mail"),
+            ("indpost_speedmail", "indpost_speedmail"),
+            ("nlpost_priority", "nlpost_priority"),
+            ("dhl_priority", "dhl_priority"),
+        ],
+        "Delivery product",
+        default="cheap",
+    )
+    print_spectrum = fields.Selection(
+        [("grayscale", "Grayscale"), ("color", "Color")],
+        default="grayscale",
+    )
+    print_mode = fields.Selection(
+        [("simplex", "Simplex"), ("duplex", "Duplex")], "Print mode", default="simplex"
+    )
+
+    push_date = fields.Datetime(readonly=True)
+    # for `error` and `pingen_error` states when we push
+    last_error_message = fields.Text("Error Message", readonly=True)
+    # pingen API v2 fields
+    pingen_uuid = fields.Char(readonly=True)
+    pingen_status = fields.Char(readonly=True)
+    # sendcenter infos
+    parsed_address = fields.Text(readonly=True)
+    cost = fields.Float(readonly=True)
+    currency_id = fields.Many2one("res.currency", "Currency", readonly=True)
+    country_id = fields.Many2one("res.country", "Country", readonly=True)
+    send_date = fields.Datetime("Date of sending", readonly=True)
+    pages = fields.Integer(readonly=True)
+    company_id = fields.Many2one(related="attachment_id.company_id")
+
+    _sql_constraints = [
+        (
+            "pingen_document_attachment_uniq",
+            "unique (attachment_id)",
+            "Only one Pingen document is allowed per attachment.",
+        ),
+    ]
+
+    def _push_to_pingen(self, pingen=None):
+        """Push a document to pingen.com
+        :param Pingen pingen: optional pingen object to reuse session
+        """
+        decoded_document = self.attachment_id._decoded_content()
+        if pingen is None:
+            pingen = self.company_id._get_pingen_client()
+        try:
+            doc_id, post_id, infos = pingen.push_document(
+                self.name,
+                StringIO(decoded_document),
```

Comment:
:red_circle:  @grindtildeath  afaik cStringIO from v10 should be replaced with io.StringIO but here it seems pdf sent to pingen is corrupted? or test attachment creation is faulty.

Reply:
Didn't you try with `io.BytesIO` ?